packages/core/src/Site/index.js

```diff
@@ -174,12 +174,13 @@ class Site {
    * @param templatePath
    */
   static initSite(rootPath, templatePath) {
-    return new Promise((resolve, reject) => {
-      new Template(rootPath, templatePath).init()
-        .then(resolve)
-        .catch((err) => {
-          reject(new Error(`Failed to initialize site with given template with error: ${err.message}`));
-        });
+    return new Promise(async (resolve, reject) => {
```

Comment:
> Is the current implementation acceptable or should I extract these functions to improve readability?

let's extract them : )

there are a few redundant wrapping promises before already, let's remove those as well (e.g. here we could make the `initSite` method async instead)

Reply:
Alright I've extracted them and remove redundant promises wrapping.